ArchiSteamFarm/Core/AprilFools.cs

```diff
@@ -0,0 +1,87 @@
+//   / _ \  | '__|/ __|| '_ \ | |\___ \ | __|/ _ \ / _` || '_ ` _ \ | |_  / _` || '__|| '_ ` _ \
+//  / ___ \ | |  | (__ | | | || | ___) || |_|  __/| (_| || | | | | ||  _|| (_| || |   | | | | | |
+// /_/   \_\|_|   \___||_| |_||_||____/  \__|\___| \__,_||_| |_| |_||_|   \__,_||_|   |_| |_| |_|
+// |
+// Copyright 2015-2021 Łukasz "JustArchi" Domeradzki
+// Contact: JustArchi@JustArchi.net
+// |
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// |
+// http://www.apache.org/licenses/LICENSE-2.0
+// |
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Globalization;
+using System.Threading;
+
+namespace ArchiSteamFarm.Core {
+	internal static class AprilFools {
+		private static readonly object LockObject = new();
+
+		// We don't care about CurrentCulture global config property, because April Fools are never initialized in this case
+		private static readonly CultureInfo OriginalCulture = CultureInfo.CurrentCulture;
+
+		private static readonly Timer Timer = new(Init);
+
+		internal static void Init(object? state = null) {
+			DateTime now = DateTime.Now;
+
+			if ((now.Month == 4) && (now.Day == 1)) {
+				try {
+					CultureInfo.DefaultThreadCurrentCulture = CultureInfo.DefaultThreadCurrentUICulture = CultureInfo.CreateSpecificCulture("qps-Ploc");
+				} catch (Exception e) {
+					ASF.ArchiLogger.LogGenericDebuggingException(e);
+
+					return;
+				}
+
+				TimeSpan aprilFoolsEnd = TimeSpan.FromDays(1) - now.TimeOfDay;
+
+				lock (LockObject) {
+					Timer.Change(aprilFoolsEnd, Timeout.InfiniteTimeSpan);
+				}
+
+				return;
+			}
+
+			try {
+				CultureInfo.DefaultThreadCurrentCulture = CultureInfo.DefaultThreadCurrentUICulture = OriginalCulture;
+			} catch (Exception e) {
+				ASF.ArchiLogger.LogGenericDebuggingException(e);
+
+				return;
+			}
+
+			// Since we already verified that it's not April Fools right now, either we're in months 1-3 before 1st April this year, or 4-12 already after the 1st April
+			DateTime nextAprilFools = new(now.Month >= 4 ? now.Year + 1 : now.Year, 4, 1, 0, 0, 0, DateTimeKind.Local);
+
+			TimeSpan aprilFoolsStart = nextAprilFools - now;
+
+			lock (LockObject) {
+				// Timer can accept only dueTimes up to 2^32 - 2
+				if (aprilFoolsStart.TotalMilliseconds >= uint.MaxValue) {
+					// Fire again in 49 days or so, when we get closer
+					Timer.Change(uint.MaxValue - 1, Timeout.Infinite);
+				} else {
+					// April 1st is soon!
+					Timer.Change(aprilFoolsStart, Timeout.InfiniteTimeSpan);
+				}
+			}
```

Comment:
Why not something like this?
```c#
uint dueTime = Math.Min(uint.MaxValue - 1, (uint) aprilFoolsStart.TotalMilliseconds);
lock (LockObject) {
	Timer.Change(dueTime, Timeout.Infinite)
}
```

Reply:
`(uint) aprilFoolsStart.TotalMilliseconds` will overflow at you. Casting it to `ulong` will work though.

Reply:
That's why I chose the words "something like".